jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/DefaultServletCombinationsTest.java

```diff
@@ -0,0 +1,191 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.ee10.servlet;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.stream.Stream;
+
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.http.HttpTester;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.LocalConnector;
+import org.eclipse.jetty.server.ResourceService;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.toolchain.test.FS;
+import org.eclipse.jetty.toolchain.test.jupiter.WorkDir;
+import org.eclipse.jetty.toolchain.test.jupiter.WorkDirExtension;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.eclipse.jetty.server.ResourceService.WelcomeMode.REDIRECT;
+import static org.eclipse.jetty.server.ResourceService.WelcomeMode.REHANDLE;
+import static org.eclipse.jetty.server.ResourceService.WelcomeMode.SERVE;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+@ExtendWith(WorkDirExtension.class)
+public class DefaultServletCombinationsTest
+{
+    public WorkDir workDir;
+
+    public Path docRoot;
+    private Server server;
+    private LocalConnector connector;
+    private ServletContextHandler context;
+
+    @BeforeEach
+    public void init() throws Exception
+    {
+        docRoot = workDir.getEmptyPathDir().resolve("docroot");
+        FS.ensureDirExists(docRoot);
+        Path subdirPath = docRoot.resolve("subdir");
+        FS.ensureDirExists(subdirPath);
+        Files.writeString(docRoot.resolve("index.html"), "Static index.html at root", UTF_8);
+        Files.writeString(docRoot.resolve("foo.welcome"), "Static foo.welcome at root", UTF_8);
+        Files.writeString(subdirPath.resolve("index.html"), "Static index.html at subdir", UTF_8);
+        Files.writeString(subdirPath.resolve("foo.welcome"), "Static foo.welcome at subdir", UTF_8);
```

Comment:
Can we have subdirs that have an index.html, but not a foo.welcome file and vice versa, also a subdir with no welcome files.

Remember we need to test the combinations: no file & no servlet; no file & servlet; file & no servlet; file & servlet

It would also be good to test the fact that a welcome with a file takes priority over a welcome servlet.